scrapy/settings/__init__.py

```diff
@@ -88,19 +103,48 @@ def getdict(self, name, default=None):
             value = json.loads(value)
         return dict(value)
 
+    def _getcomposite(self, name):
+        # DO NOT USE THIS FUNCTION IN YOUR CUSTOM PROJECTS
+        # It's for internal use in the transition away from the _BASE settings and
+        # will be removed along with _BASE support in a future release
+        basename = name + "_BASE"
+        if basename in self.attributes:
+            warnings.warn('_BASE settings are deprecated.',
+                          category=ScrapyDeprecationWarning)
+            compsett = BaseSettings(self[name + "_BASE"], priority='default')
+            compsett.update(self.get(name))
+            return compsett
+        else:
+            return self[name]
+
+    def getpriority(self, name):
+        prio = None
+        if name in self.attributes:
+            prio = self.attributes[name].priority
+        return prio
+
+    def maxpriority(self):
+        if len(self) > 0:
+            return max(self.getpriority(name) for name in self.attributes)
```

Comment:
nitpick: `for name in self` instead of `for name in self.attributes`. I think there are other similar calls that could be replaced.